.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,78 @@
+name: Prepare Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to release (e.g. 1.2.3)"
+        required: true
+        type: string
+      base_branch:
+        description: "Base branch for release (e.g. release/v1.x, hotfix/v1.2.x)"
+        required: false
+        default: "main"
+        type: string
+
+jobs:
+  prepare-release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.base_branch }}
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: "npm"
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Create draft release
+        run: gh release create v${{ inputs.version }} --draft --generate-notes
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is there a reason to create a draft release? Publishing the draft won't publish, editing the draft doesn't work either, and we have to delete the draft if we abort a release. So I only see downsides to creating a draft.